source/common/http/http2/codec_impl.h

```diff
@@ -387,7 +388,11 @@ class ConnectionImpl : public virtual Connection,
    */
   struct ServerStreamImpl : public StreamImpl, public ResponseEncoder {
     ServerStreamImpl(ConnectionImpl& parent, uint32_t buffer_limit)
-        : StreamImpl(parent, buffer_limit), headers_or_trailers_(RequestHeaderMapImpl::create()) {}
+        : StreamImpl(parent, buffer_limit), headers_or_trailers_(RequestHeaderMapImpl::create()),
+          buffer_memory_account_(std::make_shared<Buffer::BufferMemoryAccount>()) {
+      pending_recv_data_.bindAccount(buffer_memory_account_);
+      pending_send_data_.bindAccount(buffer_memory_account_);
```

Comment:
I think there are no tests covering changes to http2 codec.

Reply:
As discussed out of band, moved this to the next follow up PR which will focus on wiring this for H2 downstream + upstream.